data.sql

```diff
@@ -12,4 +12,50 @@ VALUES ('Charmander', '2020-02-08', 0, false, -11),
   ('Squirtle', '1993-04-02', 3, false, -12.13),
   ('Angemon', '2005-06-12', 1, true, -45),
   ('Boarmon', '2005-06-07', 7, true, 20.4),
-  ('Blossom', '1998-10-13', 3, true, 17);
\ No newline at end of file
+  ('Blossom', '1998-10-13', 3, true, 17);
+
+
+INSERT INTO owners (full_name, age)
+VALUES ('Sam Smith', 34),
+  ('Jennifer Orwell', 19),
+  ('Bob', 45),
+  ('Melody Pond', 77),
+  ('Dean Winchester', 14),
+  ('Jodie Whittaker', 38);
+
+INSERT INTO species (name)
+VALUES ('Pokemon'),
+  ('Digimon');
+
+UPDATE animals
+SET species_id = species.id FROM species
+WHERE animals.name LIKE '%mon' AND species.name = 'Digimon';
+
+UPDATE animals
+SET species_id = species.id FROM species
+WHERE species_id IS NULL AND species.name = 'Pokemon';
+
+UPDATE animals
+SET owner_id = owners.id FROM owners
+WHERE animals.name = 'Agumon' AND owners.full_name = 'Sam Smith';
+
+UPDATE animals
+SET owner_id = owners.id FROM owners
+WHERE animals.name = 'Gabumon' AND owners.full_name = 'Jennifer Orwell'
+  OR animals.name = 'Pikachu' AND owners.full_name = 'Jennifer Orwell';
+
+UPDATE animals
+SET owner_id = owners.id FROM owners
+WHERE animals.name = 'Devimon' AND owners.full_name = 'Bob'
+  OR animals.name = 'Plantmon' AND owners.full_name = 'Bob';
+
+UPDATE animals
+SET owner_id = owners.id FROM owners
+WHERE animals.name = 'Charmander' AND owners.full_name = 'Melody Pond'
+  OR animals.name = 'Squirtle' AND owners.full_name = 'Melody Pond'
+  OR animals.name = 'Blossom' AND owners.full_name = 'Melody Pond';
+
+UPDATE animals
+SET owner_id = owners.id FROM owners
+WHERE animals.name = 'Angemon' AND owners.full_name = 'Dean Winchester'
+  OR animals.name = 'Boarmon' AND owners.full_name = 'Dean Winchester';
```

Comment:
### Optional suggestions

- [ ] Those solution should work however, It would make your query's to be complicated, I suggest to use this way to make it quickly. 

 `UPDATE animals
   SET owner_id = (SELECT id from owners WHERE full_name = 'Sam Smith')
   WHERE name = 'Agumon'; `

Above is solution for the first question